textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexsplit/RegexSplitOperator.java

```diff
@@ -0,0 +1,185 @@
+package edu.uci.ics.textdb.dataflow.regexsplit;
+
+import edu.uci.ics.textdb.api.exception.TextDBException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import edu.uci.ics.textdb.api.common.FieldType;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.exception.ErrorMessages;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.StringField;
+import edu.uci.ics.textdb.common.field.TextField;
+import edu.uci.ics.textdb.dataflow.common.AbstractSingleInputOperator;
+
+/**
+ * @author Qinhua Huang
+ * @author Zuozhi Wang
+ * 
+ * This class is to divide an attribute by using a regex in order to get multiple tuples, 
+ * with the other attributes unchanged.
+ */
+
+public class RegexSplitOperator extends AbstractSingleInputOperator implements ISourceOperator{
+    
+    private RegexSplitPredicate predicate;
+    
+    private List<ITuple> outputTupleBuffer;
+    private int bufferCursor;
+    private boolean hasOutputBufferTuple;
+    
+    Schema inputSchema;
+    
+    
+    public RegexSplitOperator(RegexSplitPredicate predicate) { // Attribute attribute
+        
+        this.predicate = predicate;
+        this.bufferCursor = 0;
+        this.hasOutputBufferTuple = false;
+    }
+    
+    @Override
+    protected void setUp() throws DataFlowException {        
+        inputSchema = inputOperator.getOutputSchema();
+        outputSchema = inputSchema;
+    }
+    
+    @Override
+    public void open() throws TextDBException {
```

Comment:
you don't have to override`open()` function. the `open()` function in the abstract class is sufficient enough